components/director/internal/domain/application/resolver_test.go

```diff
@@ -845,6 +845,65 @@ func TestResolver_MergeApplications(t *testing.T) {
 	}
 }
 
+func TestResolver_ApplicationBySystemNumber(t *testing.T) {
+	// GIVEN
+	systemNumber := "18"
+	modelApplication := fixModelApplication("foo", "tenant-foo", appName, "Bar")
+	gqlApplication := fixGQLApplication("foo", appName, "Bar")
+
+	testCases := []struct {
```

Comment:
We can add test cases for:
- r.transact.Begin() returns error
- r.appSvc.GetBySystemNumber(ctx, systemNumber) returns error
- r.appSvc.GetBySystemNumber(ctx, systemNumber) returns NewNotFoundError()
- tx.Commit() returns error 
See similar tests for reference.